### PR TITLE
Make INodeDefManager::getIds return a vector, not a set

### PR DESCRIPTION
**Pull request by kahrl**
Change std::set to std::vector in the interface of INodeDefManager::getIds() (like already done in NodeResolver::getIdsFromNrBacklog) and everywhere that calls getIds().

This should bring performance improvements for ABMs and the Lua API functions that use getIds(): find_node_near, find_nodes_in_area, find_nodes_in_area_under_air, since std::vector is generally faster than std::set.

However, linear lookup in a vector might cause performance regressions, so the changes have to be profiled with various workloads.

The second commit fixes https://forum.minetest.net/viewtopic.php?f=6&t=13870 (ABM gets executed even if node is not longer in group). If requested, I can split it into a different PR, but in its current state it would have to depend on this one.

TODO:
- [ ] Performance measurements
- [ ] Keep ABMHandler's content_t vectors/maps around instead of re-creating them each ABM interval?

### CODE REVIEW

**Comment by nerzhul** [comment]
Maybe a typedef could be good for all those std::vector<T> with same goal

**Comment by kilbith** [comment]
@kahrl Still WIP ?

**Comment by 4aiman** [comment]
https://github.com/minetest/minetest/issues/1082

**Comment by paramat** [comment]
WIP not a priority for 0.4.16.

**Comment by nerzhul** [comment]
I'm closing this pr as ABM has been rewritten heavily to use a vector, i keep this idea in mind

**Comment by nerzhul** [comment]
Discard my comment

**Comment by paramat** [comment]
@kahrl are you able to finish this?
Closing in 1 month if no progress or adoption interest shown.

File: src/environment.cpp [hunk 1]
```diff
@@ -569,63 +570,49 @@ class ABMHandler
...
-				if(aabm.chance == 0)
+				if (aabm.chance == 0)
 					aabm.chance = 1;
 			} else {
 				aabm.chance = chance;
 			}
 			// Trigger neighbors
-			std::set<std::string> required_neighbors_s
+			std::vector<std::string> required_neighbors_s
 					= abm->getRequiredNeighbors();
```
Comment (SmallJoker) [inline src/environment.cpp]: Const ref

**Comment by nerzhul** [comment]
i may work on it, but i'm working on a second network refactor cleanup on connection.cpp/h

**Comment by paramat** [comment]
Ok, it does seem potentially useful so will keep it open.

**Comment by paramat** [comment]
Replaced by #6406

**Pull request closed without merging**